### PR TITLE
[4.2] Convert site status to service provider

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Convert the site status plugin to service provider.

### Testing Instructions
- Create three site status tasks for each type one
- Run the task
- Check the site status if it reflects the task type

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### CODE REVIEW

File: plugins/task/sitestatus/src/Extension/SiteStatus.php [hunk 1]
```diff
@@ -85,6 +78,40 @@ public static function getSubscribedEvents(): array
 		];
 	}
 
+	/**
+	 * The old config
+	 *
+	 * @var    string
```
Comment (heelc29) [inline plugins/task/sitestatus/src/Extension/SiteStatus.php]:
```suggestion
	 * @var    array
```

File: plugins/task/sitestatus/src/Extension/SiteStatus.php [hunk 2]
```diff
@@ -85,6 +78,40 @@ public static function getSubscribedEvents(): array
...
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $configFile;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param   DispatcherInterface  $dispatcher  The dispatcher
+	 * @param   array                $config      An optional associative array of configuration settings
+	 * @param   string               $oldConfig   The old config
```
Comment (heelc29) [inline plugins/task/sitestatus/src/Extension/SiteStatus.php]:
```suggestion
	 * @param   array                $oldConfig   The old config
```

File: tests/Unit/Plugin/Task/SiteStatus/SiteStatusPluginTest.php [hunk 3]
```diff
@@ -0,0 +1,243 @@
+<?php
+/**
+ * @package     Joomla.UnitTest
+ * @subpackage  Extension
+ *
+ * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Tests\Unit\Plugin\Task\SiteStatus\Extension;
```
Comment (heelc29) [inline tests/Unit/Plugin/Task/SiteStatus/SiteStatusPluginTest.php]: Please move the file to subfolder `Extension`

Comment (laoneo) [reply tests/Unit/Plugin/Task/SiteStatus/SiteStatusPluginTest.php]: done

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on 4f75fc0053f5e9dc9eb99caae28c690267a3b545<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38097">issues.joomla.org/tracker/joomla-cms/38097</a>.</sub>

File: plugins/task/sitestatus/src/Extension/SiteStatus.php [hunk 4]
```diff
@@ -113,13 +140,12 @@ public function alterSiteStatus(ExecuteTaskEvent $event): void
...
 		{
-			$offline           = self::TASKS_MAP[$event->getRoutineId()]['offline'];
-			$config['offline'] = $offline;
+			$config['offline'] = self::TASKS_MAP[$event->getRoutineId()]['offline'];
 		}
 
 		$newStatus = $config['offline'] ? 'offline' : 'online';
 		$exit      = $this->writeConfigFile(new Registry($config));
-		$this->logTask(Text::sprintf('PLG_TASK_SITE_STATUS_TASK_LOG_SITE_STATUS', $oldStatus, $newStatus));
+		$this->logTask($this->translate('PLG_TASK_SITE_STATUS_TASK_LOG_SITE_STATUS', $oldStatus, $newStatus));
```
Comment (heelc29) [inline plugins/task/sitestatus/src/Extension/SiteStatus.php:148]: Do you have read this comment? https://github.com/joomla/joomla-cms/pull/38060#discussion_r900993755

Comment (laoneo) [reply plugins/task/sitestatus/src/Extension/SiteStatus.php:148]: No didn't saw it, but we should avoid static calls for better testing.

**Comment by roland-d** [comment]
Thank you guys

**Pull request merged**